File: Frontend/src/app/models/HostingObject.ts

import { Table } from "./Table";
import { User } from "./User";

export class HostingObject
{
    public id: number;
    public name: string; 
    public adress: string; 
    public hours: string;
    public reviewStars: number;
    public phone: string;
    public mapX: number;
    public mapY: number;
    public thumbnailPicture: string;
    public tables: Table[];
    public moderator: User | null;


    constructor(Id: number, Name: string, Adress: string, Hours: string, thumbnailPicture: string ,ReviewStars: number, Phone: string, MapX: number, MapY: number, Tables :Table[], moderator : User | null)
    {
        this.id = Id;
        this.name = Name;
        this.adress = Adress;
        this.hours = Hours;
        this.reviewStars = ReviewStars;
        this.phone = Phone;
        this.mapX = MapX;
        this.mapY = MapY;
        this.tables = Tables;
        this.thumbnailPicture = thumbnailPicture;
        this.moderator = moderator;
    }
}

File: Backend/NaissusEvents/Models/HostingObject.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;



#nullable disable

namespace Models
{
  
  public class HostingObject
  {
  

    [Key]
    public int Id { get; set; }
    public string Name { get; set; }

    public string Adress { get; set; }

    public string Hours{get;set;}

    public double? ReviewStars{get;set;}  

   
    [StringLength(13, MinimumLength = 6)]
    [Phone]
    public string Phone { get; set; }

    public float MapX{get;set;}
    public float MapY{get;set;}

  
    public ICollection<Event> Events { get; set; }

   
    public ICollection<Table> Tables { get; set; }

    public ICollection<PicturesData> Pictures { get; set; }

    public string thumbnailPicture { get ; set;}

    [ForeignKey("AppUser")]
    public AppUser moderator {get; set;}

 

 

   
   
   

  
        
  }
}
